### PR TITLE
Fix #18593 - drop db line included in server export if exporting only data

### LINKED ISSUE #18593: switching between exporting data and structure, has strange effects if something in "Object creation options" is touched
In the latest versions, switching between exporting data and structure, has strange effects if something in "Object creation options" is touched for example:

If I put all the object creation options WITH, and then decide that I don't want to create the tables, I just want to export the data, this is the result:

```
DROP DATABASE IF EXISTS `DBClients`;
CREATE DATABASE IF NOT EXISTS `DBClients` DEFAULT CHARACTER SET utf8mb3 COLLATE utf8mb3_general_ci;
USE `DBClients`;
```

As you can see, despite having changed to the option of only exporting data at the last moment, it continues adding statements about the structure of the tables.

If instead of this, I don't touch anything about the creation of objects and go directly to export only data, the header is a bit different:

```
CREATE DATABASE IF NOT EXISTS DBClients DEFAULT CHARACTER SET utf8mb3 COLLATE utf8mb3_general_ci;
USE DBClients;
```

It doesn't seem right to me that versions 1 and 2, in which the change was made to export only data, have different headers, it's as if even though the object creation boxes are hidden, they still have effects in the final ".sql" script.

steps to reproduce it, scenario 1:

- at startup /phpmyadmin!!!
- go to export.
- exported custom.
- sql format.
- select the databases.
- go to Object Creation Options | Add Statements: Select All.
- go back to the top option and choose "data"
- download file #-1

steps to reproduce it, scenario 2:

- at startup /phpmyadmin!!!
- go to export.
- exported custom.
- sql format.
- select the databases.
- ~~go to Object Creation Options | Add Statements: Select All.~~ **don't do this**
- go to option and choose "data"
- download file #-2

compare file #-1 and #-2

**_I find it strange that no one had noticed this situation..._**

### PR DESCRIPTION
**Pull request by LHofman**
Signed-off-by: Lennert Hofman <hofmanlennert@gmail.com>

### Description

When doing an SQL server export, if only exporting data, don't include the "drop database" line.

Fixes #18593

Before submitting pull request, please review the following checklist:

- [ ] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [ ] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [ ] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [ ] Every commit has a descriptive commit message.
- [ ] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [ ] Any new functionality is covered by tests.

### CODE REVIEW

**Comment by LHofman** [comment]
I feel like the fix is not in the ideal situation. I would prefer to do it much earlier on so the problem won't come back for future options. Something that checks if only export data is selected that sets all structure options back to the default.

Also, I did add the sign-off to the commit message via rebase, maybe that's why it didn't find it.

File: src/Plugins/Export/ExportSql.php [hunk 1]
```diff
@@ -803,7 +803,11 @@ public function exportDBCreate(string $db, string $exportType, string $dbAlias =
 
         $compat = $GLOBALS['sql_compatibility'] ?? 'NONE';
 
-        if (isset($GLOBALS['sql_drop_database'])) {
+        $exportStructure = (
```
Comment (kamil-tekiela) [inline src/Plugins/Export/ExportSql.php:806]: Just commenting on code style: the parenthese seem unnecessary and `in_array` should always use strict mode, especially if you compare it to string constants.

Comment (LHofman) [reply src/Plugins/Export/ExportSql.php:806]: Can you elaborate on the parentheses being unnecessary?
Would you put this condition on 1 line?

File: src/Plugins/Export/ExportSql.php [hunk 2]
```diff
@@ -803,7 +803,11 @@ public function exportDBCreate(string $db, string $exportType, string $dbAlias =
...
+            ! isset($GLOBALS['sql_structure_or_data'])
+            || in_array($GLOBALS['sql_structure_or_data'], ['structure', 'structure_and_data'])
+        );
```
[the remark this answers is not in the document]
Comment (LHofman) [reply src/Plugins/Export/ExportSql.php:809]: Huh. I haven't seen this syntax used before.

[the remark this answers is not in the document]
Comment (kamil-tekiela) [reply src/Plugins/Export/ExportSql.php]: What syntax are you referring to?

[the remark this answers is not in the document]
Comment (LHofman) [reply src/Plugins/Export/ExportSql.php]: Putting conditions on multiple lines without parenthesis. But it's fine, I'll follow the syntax

**Comment by LHofman** [comment]
I have created a new PR to the correct branch: https://github.com/phpmyadmin/phpmyadmin/pull/19031

**Pull request closed without merging**